runtime/src/lib.rs

```diff
@@ -1859,18 +1947,48 @@ mod tests {
     }
 
     #[test]
+    #[should_panic(expected = "blah")]
     fn test_deterministic_panic_aborts_root() {
         let runner = deterministic::Runner::default();
         test_panic_aborts_root(runner);
     }
 
+    #[test]
+    #[should_panic(expected = "blah")]
+    fn test_deterministic_panic_aborts_root_caught() {
```

Comment:
We can't catch the root task.

Reply:
https://github.com/commonwarexyz/monorepo/pull/1800#discussion_r2403367364
naming is a bit confusing but I get it

Reply:
`..._root_catch_fails` maybe?